src/Storefront/Controller/AuthController.php

```diff
@@ -358,4 +385,49 @@ public function resetPassword(RequestDataBag $data, SalesChannelContext $context
 
         return $this->redirectToRoute('frontend.account.profile.page');
     }
+
+    /**
+     * @Since("6.0.0.0")
+     * @Route("/account/login/customer/{token}/{salesChannelId}/{customerId}", name="frontend.account.login.customer", methods={"GET"})
+     */
+    public function loginAsCustomer(string $token, string $salesChannelId, string $customerId, SalesChannelContext $context, Request $request): Response
```

Comment:
The logic itself should be in a store api route. This should be interesting also for Headless users :)